web_poet/mixins.py

```diff
@@ -1,11 +1,48 @@
+import abc
 from urllib.parse import urljoin
 
 import parsel
 from w3lib.html import get_base_url
 
 
-class ResponseShortcutsMixin:
+class SelectableMixin(abc.ABC):
+    """
+    Inherit from this mixin, implement ``._selector_input`` method,
+    get ``.selector`` property and ``.xpath`` / ``.css`` methods.
+    """
+    __cached_selector = None
+
+    @abc.abstractmethod
+    def _selector_input(self) -> str:
+        raise NotImplementedError()  # pragma: nocover
+
+    @property
+    def selector(self) -> parsel.Selector:
+        """Cached instance of :external:class:`parsel.selector.Selector`."""
+        # XXX: caching is implemented in a manual way to avoid issues with
+        # non-hashable classes, where memoizemethod_noargs doesn't work
+        if self.__cached_selector is not None:
+            return self.__cached_selector
+        # XXX: should we pass base_url=self.url, as Scrapy does?
```

Comment:
I think we can remove this comment since it's being used by `BrowserHtml` which doesn't rely on a **url**. Or do you foresee a need for it later on @kmike ?

Reply:
If that's a right thing to do (which I'm not sure about - it seems it's not needed), we'd need to have URL for selectors to work properly. In this case, having a class like BrowserResponse, which contains both URL and HTML (similar to what we had with AutoextractHtml), might be better.

That said, it won't be a part of BrowserHtml, so it does make sense to remove the comment, thanks @BurnzZ!

Reply:
Actually I think it might be better to keep the comment, as it's a part of SelectableMixin, not of BrowserHtml class.